dragula.js

```diff
@@ -551,16 +551,32 @@ function getElementBehindPoint (point, x, y) {
   var state = p.className;
   var el;
   p.className += ' gu-hide';
-  el = doc.elementFromPoint(x, y);
+  el = elementFromPoint(x, y);
   p.className = state;
   return el;
+
+  // Find the topmost element at point x, y.
+  // Traverses any shadow root to find the deepest element.
+  function elementFromPoint(x, y) {
+    var el = doc.elementFromPoint(x, y);
+    var maybeElem;
+    while (el && el.shadowRoot) {
+      maybeElem = el.shadowRoot.elementFromPoint(x, y);
+      if (maybeElem) {
+        el = maybeElem;
+      } else {
+        return el;
+      }
+    }
+    return el;
+  }
 }
 
 function never () { return false; }
 function always () { return true; }
 function getRectWidth (rect) { return rect.width || (rect.right - rect.left); }
 function getRectHeight (rect) { return rect.height || (rect.bottom - rect.top); }
-function getParent (el) { return el.parentNode === doc ? null : el.parentNode; }
+function getParent (el) { return el.parentNode === doc ? null : (el.host || el.parentNode); }
```

Comment:
Heads up, this will return an unexpected result when `el` is an `<a>` tag, due to the `host` property meaning something else on these tags. I am unsure how to get the shadow host property of an `<a>` tag.

Reply:
Hm, according to v1 specs , `el.host` is the right way to do this, maybe a check for dom node or `ShadowRoot` would sove the the problem here?